salt/cloud/clouds/azurearm.py

```diff
@@ -1834,3 +1834,63 @@ def delete_blob(call=None, kwargs=None):  # pylint: disable=unused-argument
 
     storageservice.delete_blob(kwargs['container'], kwargs['blob'])
     return True
+
+
+def stop(name, resource_group=None, call=None):
+    '''
+    Stop a VM
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+         salt-cloud -a stop myminion
+    '''
+    if call == 'function':
+        raise SaltCloudSystemExit(
+            'The stop action must be called with -a or --action.'
+        )
+    if not compconn:
+        compconn = get_conn()
+    if resource_group is None:
+        for group in list_resource_groups():
+            try:
+                instance_stop = compconn.virtual_machines.deallocate(group, name)
+                instance_stop.wait()
+                resource_group = group
```

Comment:
What is the purpose of setting this resource_group variable? It looks like it's not being used anywhere later.